.buildkite/integration.pipeline.yml

```diff
@@ -5,27 +5,77 @@ env:
   VAULT_PATH: "kv/ci-shared/observability-ingest/cloud/gcp"
 
 steps:
-  - label: "Integration tests: packaging"
-    key: "package-it"
-    command: ".buildkite/scripts/steps/integration-package.sh"
-    artifact_paths:
-      - build/distributions/**
-    agents:
-      provider: "gcp"
-      machineType: "n1-standard-8"
+  - group: "Integration tests: packaging"
+    key: "int-packaging"
+    steps:
+      - label: "Packaging: Ubuntu x86_64"
+        key: "packaging-ubuntu-x86-64"
+        env:
+          PACKAGES: "tar.gz"
+          PLATFORMS: "linux/amd64"
+        command: ".buildkite/scripts/steps/integration-package.sh"
+        artifact_paths:
+        - build/distributions/**
+        agents:
+          provider: "gcp"
+          machineType: "n1-standard-8"
+
+      - label: "Packaging: Ubuntu arm64"
+        key: "packaging-ubuntu-arm64"
+        env:
+          PACKAGES: "tar.gz"
+          PLATFORMS: "linux/arm64"
+        command: ".buildkite/scripts/steps/integration-package.sh"
+        artifact_paths:
+          - build/distributions/**
+        agents:
+          provider: "gcp"
+          machineType: "n1-standard-8"
+
+      - label: "Packaging: Windows"
+        key: "packaging-windows"
+        env:
+          PACKAGES: "zip"
+          PLATFORMS: "windows/amd64"
+        command: ".buildkite/scripts/steps/integration-package.sh"
+        artifact_paths:
+          - build/distributions/**
+        agents:
+          provider: "gcp"
+          machineType: "n1-standard-8"
+
+      - label: "Packaging: Containers {{matrix.ext}} {{matrix.arch}}"
+        key: "packaging-containers"
+        env:
+          PACKAGES: "{{matrix.ext}}"
+          PLATFORMS: "{{matrix.arch}}"
+        command: ".buildkite/scripts/steps/integration-package.sh"
+        artifact_paths:
+          - build/distributions/**
+        agents:
+          provider: "gcp"
+          machineType: "n1-standard-8"
+        matrix:
+          setup:
+            arch:
+              - linux/amd64
+              - linux/arm64
+            ext:
+              - rpm
+              - deb
 
   - label: "Serverless integration test"
     key: "serverless-integration-tests"
     depends_on:
-      - package-it
+      - int-packaging
     concurrency_group: elastic-agent-extended-testing/serverless-integration
     concurrency: 8
     env:
       # we run each step in a different data center to spread the load
       TEST_INTEG_AUTH_GCP_DATACENTER: "us-central1-a"
-    command: |
-      buildkite-agent artifact download build/distributions/** . --step 'package-it'
-      .buildkite/scripts/steps/integration_tests.sh serverless integration:single TestLogIngestionFleetManaged #right now, run a single test in serverless mode as a sort of smoke test, instead of re-running the entire suite
+    commands:
+      - buildkite-agent artifact download "build/distributions/**" . $BUILDKITE_BUILD_ID
```

Comment:
any reason we are we moving away from the block scalar (`|`) to arrays? Is the reason that you want to rely on buildkite's built-in failure handling (

We've established the pattern with `|` in other repos which IMHO enhances readability and helps with copying/pasting for reproductions which why I am bringing this up.